store.go

```diff
@@ -0,0 +1,226 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the LGPL, see LICENCE file for details.
+
+package aclstore
+
+import (
+	"context"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/juju/simplekv"
+	"gopkg.in/errgo.v1"
+)
+
+var ErrACLNotFound = errgo.Newf("ACL not found")
+
+// separator is used as the character to divide usernames in the ACL.
+// This needs to be a character that's illegal in usernames.
+const separator = " "
+
+// ACLStore is the persistent storage interface used by an ACLHandler.
+type ACLStore interface {
+	// NewACL creates an ACL with the given name and initial users.
+	// If the ACL already exists, this is a no-op and the initialUsers
+	// argument is ignored.
+	NewACL(ctx context.Context, aclName string, initialUsers ...string) error
```

Comment:
As discussed I have some misgivings about this being part of the interface, but we shall see how it progresses before making judgement.